### PR TITLE
Add filter to job subscription request

### LINKED ISSUE #39: Search filter is not applied to new jobs
When you add a filter in the search bar of the jobs list the jobs are filtered accordingly. However, new jobs that come in via event stream are added regardless of the filter.

**Steps to reproduce:**
- Add a filter to the search bar.
- Create a job that does not meet the filter criteria.
- → The job is added to the job list anyway.

**Expected behavior:**
- The job is not added since it does not meet the filter criteria.

### PR DESCRIPTION
**Pull request by corneliusludmann**
With this commit the search filter is applied to the job subscription event stream as well.

_Feel free to improve this according to your needs._

Fixes #39.

### CODE REVIEW

File: pkg/webui/src/JobList.tsx [hunk 1]
```diff
@@ -121,6 +127,9 @@ class JobListImpl extends React.Component<JobListProps, JobListState> {
         const resp = await new Promise<ListJobsResponse>((resolve, reject) => this.props.client.listJobs(req, (err, resp) => !!err ? reject(err) : resolve(resp!)));
         state.jobs = resp.getResultList().map(r => r.toObject());
         state.totalJobs = resp.getTotal();
+        if (newState.search !== this.state.search) {
+            this.startListening();
+        }
```
Comment (csweichel) [inline pkg/webui/src/JobList.tsx]: `startListening` uses the state to determine the new search criteria.
We'd need to call `startListening` once the state has been set, e.g.

```TypeScript
const searchChanged = newState.search !== this.state.search;
this.setState(state);

if (searchChanged) {
    this.startListening();
}
```

File: pkg/webui/src/JobList.tsx [hunk 2]
```diff
@@ -68,9 +70,13 @@ class JobListImpl extends React.Component<JobListProps, JobListState> {
     protected startListening() {
         try {
             const req = new SubscribeRequest();
-            let evts = this.props.client.subscribe(req);
-            evts.on('end', () => setTimeout(() => this.startListening(), 1000));
-            evts.on('data', r => {
+            req.setFilterList(this.state.search);
+            if (this.eventStream) {
+                this.eventStream.cancel();
+            }
```
Comment (csweichel) [inline pkg/webui/src/JobList.tsx]: nit: I'd cancel the stream first and then create the request (i.e. move line 73 behind the if). This way request creation and use are closer together.

File: pkg/webui/src/JobList.tsx (hunk 1 above)
Comment (corneliusludmann) [reply pkg/webui/src/JobList.tsx]: You're right. I'm surprised it worked in my tests anyway!? Gonna change this later ...

I wonder how useful the search comparison is at all? If I understand it correctly it doesn't perform a deep equal, does it? Thus, the only case when it is equal is when both are `undefined`? Right? However, probably better than nothing ...

**Comment by csweichel** [comment]
Thank you very much for the fix/PR.
I got impatient, have merged your fix and implemented the comments myself.

**Comment by corneliusludmann** [comment]
> I got impatient, have merged your fix and implemented the comments myself.

Yeah, great. I'm glad it's fixed. :-)

**Pull request merged**